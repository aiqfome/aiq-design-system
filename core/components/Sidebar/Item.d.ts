import React from 'react';
interface ItemProps {
    item: any;
    openItem: boolean;
    sidebarOpened: boolean;
    toggleItem: () => void;
    heightScrolledToTop?: number;
}
export declare const Item: React.FC<ItemProps>;
export {};
//# sourceMappingURL=Item.d.ts.map