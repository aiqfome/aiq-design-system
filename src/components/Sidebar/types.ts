export type ItemType = "internal" | "external";

type SubItemProps = {
    name: string
    href: string
    badge?: number
  }

export type ItemObjectProps = {
    name: string
    
    href?: string
    active?: boolean
    icon?: JSX.Element
    exact?: boolean
    items?: Array<SubItemProps>
    callback?: any
    type?: ItemType
  }