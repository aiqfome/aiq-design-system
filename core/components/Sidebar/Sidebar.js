"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Sidebar = void 0;
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importDefault(require("styled-components"));
const Divider_1 = require("../Divider");
const Flex_1 = require("../Flex");
const Item_1 = require("./Item");
const Items = styled_components_1.default.ul `
  list-style: none;
  padding: 0;
  margin: 0;
  overflow-y: auto;
  overflow-x: hidden;
  direction: ltr;

  &::-webkit-scrollbar {
    width: 5px;
  }

  &::-webkit-scrollbar-track {
    background-color: ${({ theme }) => theme.colors.white};
    -webkit-border-radius: 10px;
    border-radius: 10px;
  }

  &::-webkit-scrollbar-thumb {
    -webkit-border-radius: 10px;
    border-radius: 10px;
    background: ${({ theme }) => theme.colors.mediumGrey};
  }
`;
const SidebarStyled = styled_components_1.default(Flex_1.Flex) `
  box-shadow: 0px 3px 15px #0000001a;
  height: 100vh;
  max-width: ${({ opened }) => (opened ? '340px' : '60px')};
  transition: all 0.5s ease;
  will-change: transform;
  position: fixed;
`;
exports.Sidebar = (_a) => {
    var { data, header, opened = false } = _a, props = __rest(_a, ["data", "header", "opened"]);
    const [openItem, setOpenItem] = react_1.useState('');
    const [heightScrolledToTop, setHeightScrolledToTop] = react_1.useState(0);
    react_1.useEffect(() => {
        function listenWhenSidebarScroll(event) {
            setHeightScrolledToTop(event.target.scrollTop);
        }
        const elementItensSidebar = document.querySelectorAll('#items-sidebar')[0];
        elementItensSidebar.addEventListener('scroll', listenWhenSidebarScroll);
        return () => {
            elementItensSidebar.removeEventListener('scroll', listenWhenSidebarScroll);
        };
    }, []);
    const toggleItem = (value = '') => {
        setOpenItem(openItem === value ? '' : value);
    };
    return (react_1.default.createElement(SidebarStyled, Object.assign({ width: '100%', backgroundColor: '#fff', maxWidth: '340px', flexDirection: 'column', opened: opened, "data-testid": 'sidebar', className: opened ? 'show' : 'hidden' }, props), data.length >= 0 && (react_1.default.createElement(react_1.default.Fragment, null,
        header && header,
        react_1.default.createElement(Divider_1.Divider, { width: '100%', marginBottom: '16px' }),
        react_1.default.createElement(Items, { id: 'items-sidebar' }, data.map((item, index) => (react_1.default.createElement(Item_1.Item, { key: index, item: item, sidebarOpened: opened, openItem: openItem === index, toggleItem: () => toggleItem(index), heightScrolledToTop: heightScrolledToTop }))))))));
};
exports.Sidebar.propTypes = {
    data: prop_types_1.default.any.isRequired,
    opened: prop_types_1.default.bool,
    header: prop_types_1.default.any
};
