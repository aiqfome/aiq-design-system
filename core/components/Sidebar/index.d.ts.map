{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/Sidebar/index.tsx"],"names":[],"mappings":"AAAA,cAAc,WAAW,CAAA"}