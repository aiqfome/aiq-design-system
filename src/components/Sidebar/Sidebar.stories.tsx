import React from 'react'
import { MdHome, MdBookmark, MdStorage, MdExitToApp } from 'react-icons/md'
import { BrowserRouter as Router } from 'react-router-dom'
import { Sidebar } from './Sidebar'
import { withKnobs, boolean } from '@storybook/addon-knobs'

import { Flex } from '../Flex'
import { Avatar } from '../Avatar'

export default {
  component: Sidebar,
  title: 'Sidebar',
  decorators: [withKnobs]
}

const sidebarData = {
  user: {
    name: 'juliano',
    email: 'test@aiqfome.com'
  },
  itens: [
    {
      icon: <MdHome size={18} />,
      name: 'dashboard',
      href: '/dashboard',
      badge: 1
    },
    {
      icon: <MdBookmark size={18} />,
      name: 'bookmark',
      href: '/pedidos',
      active: true
    },
    {
      icon: <MdBookmark size={18} />,
      name: 'bookmark',
      href: '/pedidos',
      active: true
    },
    {
      icon: <MdBookmark size={18} />,
      name: 'bookmark',
      href: '/pedidos',
      active: true
    },
    {
      icon: <MdBookmark size={18} />,
      name: 'bookmark',
      href: '/pedidos',
      active: true
    },
    {
      icon: <MdBookmark size={18} />,
      name: 'bookmark',
      href: '/pedidos',
      active: true
    },
    {
      icon: <MdBookmark size={18} />,
      name: 'bookmark',
      href: '/pedidos',
      active: true
    },
    {
      icon: <MdBookmark size={18} />,
      name: 'bookmark',
      href: '/pedidos',
      active: true
    },
    {
      icon: <MdBookmark size={18} />,
      name: 'bookmark',
      href: '/pedidos',
      active: true
    },
    {
      icon: <MdBookmark size={18} />,
      name: 'bookmark',
      href: '/pedidos',
      active: true
    },
    {
      icon: <MdBookmark size={18} />,
      name: 'bookmark',
      href: '/pedidos',
      active: true
    },
    {
      icon: <MdStorage size={18} />,
      name: 'store',
      itens: [
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem',
          badge: 18
        }
      ]
    },
    {
      icon: <MdStorage size={18} />,
      name: 'itens',
      itens: [
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item',
          href: '/subItem'
        },
        {
          name: 'Sub Item END',
          href: '/subItem'
        }
      ]
    },
    {
      icon: <MdBookmark size={18} />,
      name: 'aiqfome',
      href: 'https://aiqfome.com/',
      type: 'external'
    },
    {
      icon: <MdExitToApp size={18} />,
      name: 'sair',
      callback: () => {
        console.log('logout')
      }
    }
  ]
}

const Header = () => {
  const user = sidebarData.user
  return (
    <Flex
      style={{
        overflowX: 'hidden'
      }}
      py='8px'
      alignItems='center'
    >
      <Avatar alt={user.name} mx='12px' />
    </Flex>
  )
}

export const Basic: React.FC = () => {
  return (
    <Router>
      <Flex backgroundColor='#E1E1E1' justifyContent='row'>
        <Sidebar
          header={<Header />}
          opened={boolean('opened', true)}
          data={sidebarData}
        />
        <h1>test</h1>
      </Flex>
    </Router>
  )
}
