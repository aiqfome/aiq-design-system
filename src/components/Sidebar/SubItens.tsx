import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import { Text } from '../Text'
import { Flex } from '../Flex'
import { Link } from '../Link'
import { Badge } from '../Badge'

interface Props {
  item?: any
  itemOpened?: boolean
  sidebarOpened?: boolean
  heightScrolledToTop?: number
}

const SubItensStyled = styled(Flex)<Props>`
  ul {
    margin: 0;
    padding: 0;
    list-style: none;
  }

  ${({ sidebarOpened }) => {
    if (sidebarOpened) {
      return css`
        background: #f6f6f6;

        &.hide {
          overflow: hidden;
          max-height: 0;
          padding-top: 0;
          padding-bottom: 0;
          margin-top: 0;
          margin-bottom: 0;
          transition: all 0.2s ease;
          will-change: transform;
        }

        &.show {
          max-height: 100vh;
          overflow: hidden;
          transition: all 0.5s ease;
          will-change: transform;
        }

        a {
          color: #434343;
          padding: 12px 24px;
          display: flex;
          flex-direction: row;
          justify-content: space-between;

          &:hover {
            background: ${({ theme }) => theme.colors.grey};
          }
        }
      `
    }

    return css<Props>`
      a {
        font-size: 14px;
        color: #434343;
        padding: 6px 0px;
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        &:hover {
          background: ${({ theme }) => theme.colors.lightGrey};
        }
      }

      ${({ itemOpened, heightScrolledToTop }) => {
        if (itemOpened) {
          return css`
            display: flex;
            position: fixed;
            width: 240px;
            margin-left: 55px;
            margin-top: calc((50px + ${heightScrolledToTop}px) * -1);
            background: #ffff;
            border-top-right-radius: 4px;
            border-bottom-right-radius: 4px;
            box-shadow: 2px 3px 4px #00000029;
          `
        }
        return css`
          display: none;
        `
      }}
    `
  }}
`

export const SubItens = ({
  item,
  sidebarOpened,
  itemOpened,
  heightScrolledToTop = 0
}) => {
  return (
    <SubItensStyled
      heightScrolledToTop={heightScrolledToTop}
      flexDirection='column'
      className={`${itemOpened ? 'show' : 'hide'}`}
      sidebarOpened={sidebarOpened}
      itemOpened={itemOpened}
    >
      {!sidebarOpened && (
        <Text color='primary' mx={6} mt={6} mb={4} fontWeight='semiBold'>
          {item.name}
        </Text>
      )}
      {item.itens && (
        <ul>
          {item.itens.map((subItem, index) => (
            <li key={index}>
              <Link
                variant={subItem.type ? subItem.type : 'internal'}
                href={subItem.href}
              >
                <Text
                  cursor='pointer'
                  fontSize='medium'
                  color='darkGrey'
                  px={6}
                  my={2}
                >
                  {subItem.name}
                </Text>

                {subItem.badge && <Badge count={subItem.badge} />}
              </Link>
            </li>
          ))}
        </ul>
      )}
    </SubItensStyled>
  )
}

SubItens.propTypes = {
  item: PropTypes.any,
  itemOpened: PropTypes.bool,
  sidebarOpened: PropTypes.bool,
  heightScrolledToTop: PropTypes.number
}
