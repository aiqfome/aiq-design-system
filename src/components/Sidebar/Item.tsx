import React, { useState, useMemo } from 'react'
import PropTypes from 'prop-types'
import { useLocation } from 'react-router-dom'
import styled, { css } from 'styled-components'

import { MdExpandMore, MdExpandLess } from 'react-icons/md'

import { Text } from '../Text'
import { Flex } from '../Flex'
import { Link } from '../Link'
import { Icon } from '../Icon'
import { Badge } from '../Badge'

import { SubItens } from './SubItens'

interface ItemProps {
  item: any
  sidebarOpened: boolean
  heightScrolledToTop?: number
}

interface ItemStyledProps {
  sidebarOpened?: boolean
  active?: boolean
}

const ItemStyled = styled.li<ItemStyledProps>`
  position: relative;
  transition: 0.3ms;

  ${({ theme, active }) =>
    active &&
    css`
      svg {
        color: ${theme.colors.primary};
      }
    `}

  &:hover {
    cursor: pointer;

    ${({ theme, sidebarOpened }) =>
      !sidebarOpened &&
      css`
        svg {
          color: ${theme.colors.primary};
        }
      `};

    background: ${({ theme, sidebarOpened }) =>
      sidebarOpened ? theme.colors.lightGrey : theme.colors.white};
  }
`

const LinkStyled = styled(Link)`
  padding: 16px 22px;
  display: flex;
  flex-direction: row;
`

export const Item: React.FC<ItemProps> = ({
  item,
  sidebarOpened = false,
  heightScrolledToTop,
  ...props
}) => {
  const [isOpen, setIsOpen] = useState(false)
  const location = useLocation()

  function computeBadgeAllItens(item) {
    let value = 0

    if (item.itens) {
      item.itens.forEach(subItem => {
        if (subItem.badge) value += subItem.badge
      })
    }

    if (item.badge) {
      value += item.badge
    }

    return value
  }

  const badgeAllItens = useMemo(() => computeBadgeAllItens(item), [item])

  function changeVisibilitySubItem() {
    if (sidebarOpened) {
      setIsOpen(!isOpen)
    }
  }

  function isItemActive() {
    let active = false
    if (item.href) {
      active = location.pathname.includes(item.href)
      if (item.exact) active = location.pathname === item.href
    }
    item.itens &&
      item.itens.forEach(subItem => {
        if (!active) {
          active = location.pathname.includes(subItem.href)
          if (item.exact) active = location.pathname === subItem.href
        }
      })
    return active
  }

  const ItemWrapper = ({ children }) => {
    if (item.itens && item.itens.length > 0) {
      return (
        <Flex
          flexDirection='row'
          width='100%'
          alignItems='center'
          justifyContent='space-between'
          padding='16px 22px'
          onClick={changeVisibilitySubItem}
        >
          {children}
        </Flex>
      )
    }
    if (item.callback) {
      return (
        <Flex
          flexDirection='row'
          width='100%'
          alignItems='center'
          justifyContent='space-between'
          padding='16px 22px'
          onClick={item.callback}
        >
          {children}
        </Flex>
      )
    }
    return (
      <LinkStyled variant={item.type ? item.type : 'internal'} href={item.href}>
        {children}
      </LinkStyled>
    )
  }
  ItemWrapper.propTypes = {
    children: PropTypes.any
  }

  return (
    <ItemStyled
      active={isItemActive()}
      sidebarOpened={sidebarOpened}
      onMouseEnter={() => !sidebarOpened && setIsOpen(true)}
      onMouseLeave={() => !sidebarOpened && setIsOpen(false)}
    >
      <ItemWrapper>
        <Icon marginRight='22px' color={sidebarOpened ? 'primary' : 'grey'}>
          {item.icon}
        </Icon>

        {sidebarOpened && (
          <Flex
            flex={1}
            justifyContent='space-between'
            onClick={changeVisibilitySubItem}
          >
            <Flex flex={1}>
              <Text cursor='pointer' color='darkerGrey'>
                {item.name}
              </Text>
            </Flex>

            {badgeAllItens > 0 && !isOpen && <Badge count={badgeAllItens} />}

            {item.itens && (
              <Icon color='primary'>
                {isOpen ? (
                  <MdExpandLess size={18} />
                ) : (
                  <MdExpandMore size={18} />
                )}
              </Icon>
            )}
          </Flex>
        )}
      </ItemWrapper>
      <SubItens
        item={item}
        heightScrolledToTop={heightScrolledToTop}
        sidebarOpened={sidebarOpened}
        itemOpened={isOpen}
      />
    </ItemStyled>
  )
}

Item.propTypes = {
  item: PropTypes.any.isRequired,
  sidebarOpened: PropTypes.bool.isRequired,
  heightScrolledToTop: PropTypes.number
}
