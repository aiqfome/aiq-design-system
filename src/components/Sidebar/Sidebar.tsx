import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Divider } from '../Divider'
import { Flex } from '../Flex'
import { Item } from './Item'

export interface Props {
  data: { itens: any }
  opened?: boolean
  header?: any
}

const Itens = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;
  overflow-y: auto;
  direction: ltr;

  &::-webkit-scrollbar {
    width: 5px;
  }

  &::-webkit-scrollbar-track {
    background-color: #ebebeb;
    -webkit-border-radius: 10px;
    border-radius: 10px;
  }

  &::-webkit-scrollbar-thumb {
    -webkit-border-radius: 10px;
    border-radius: 10px;
    background: ${({ theme }) => theme.colors.primaryLight};
  }
`

interface SidebarStyledProps {
  opened?: boolean
}

const SidebarStyled = styled(Flex)<SidebarStyledProps>`
  box-shadow: 0px 3px 15px #0000001a;
  height: 100vh;
  max-width: ${({ opened }) => (opened ? '340px' : '60px')};
  transition: all 0.5s ease;
  will-change: transform;
  position: fixed;
`

export const Sidebar: React.FC<Props> = ({
  data,
  header,
  opened = false,
  ...props
}) => {
  const [heightScrolledToTop, setHeightScrolledToTop] = useState(0)
  useEffect(() => {
    function listenWhenSidebarScroll(event) {
      setHeightScrolledToTop(event.target.scrollTop)
    }
    const elementItensSidebar = document.querySelectorAll('#itens-sidebar')[0]
    elementItensSidebar.addEventListener('scroll', listenWhenSidebarScroll)
    return () => {
      elementItensSidebar.removeEventListener('scroll', listenWhenSidebarScroll)
    }
  }, [])

  return (
    <SidebarStyled
      width='100%'
      backgroundColor='#fff'
      maxWidth='340px'
      flexDirection='column'
      opened={opened}
      {...props}
    >
      {data && (
        <>
          {header && header}
          <Divider width='100%' marginBottom='16px' />
          <Itens id='itens-sidebar'>
            {data.itens.map((item, index) => (
              <Item
                heightScrolledToTop={heightScrolledToTop}
                sidebarOpened={opened}
                key={index}
                item={item}
              />
            ))}
          </Itens>
        </>
      )}
    </SidebarStyled>
  )
}

Sidebar.propTypes = {
  data: PropTypes.any.isRequired,
  opened: PropTypes.bool,
  header: PropTypes.any
}
