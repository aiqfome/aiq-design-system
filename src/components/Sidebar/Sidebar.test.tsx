import React from 'react'

import { MdHome, MdBookmark, MdStorage } from 'react-icons/md'
import { BrowserRouter as Router } from 'react-router-dom'

import { render } from '../utils/test/render'
import { Sidebar } from './Sidebar'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const sidebarData = {
      user: {
        name: 'juliano',
        email: 'test@aiqfome.com'
      },
      itens: [
        {
          icon: <MdHome size={18} />,
          name: 'dashboard',
          href: '/dashboard'
        },
        {
          icon: <MdBookmark size={18} />,
          name: 'bookmark',
          href: '/pedidos'
        },
        {
          icon: <MdStorage size={18} />,
          name: 'store',
          itens: [
            {
              name: 'Sub Item',
              href: '/subItem'
            },
            {
              name: 'Sub Item',
              href: '/subItem'
            },
            {
              name: 'Sub Item',
              href: '/subItem',
              badge: 18
            }
          ]
        },
        {
          icon: <MdStorage size={18} />,
          name: 'itens',
          itens: [
            {
              name: 'Sub Item',
              href: '/subItem'
            },
            {
              name: 'Sub Item',
              href: '/subItem'
            },
            {
              name: 'Sub Item',
              href: '/subItem'
            },
            {
              name: 'Sub Item',
              href: '/subItem'
            },
            {
              name: 'Sub Item',
              href: '/subItem'
            },
            {
              name: 'Sub Item',
              href: '/subItem'
            }
          ]
        },
        {
          icon: <MdBookmark size={18} />,
          name: 'aiqfome',
          href: 'https://aiqfome.com/',
          type: 'external'
        }
      ]
    }

    const component = render(
      <Router>
        <Sidebar opened={true} data={sidebarData} />
      </Router>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
