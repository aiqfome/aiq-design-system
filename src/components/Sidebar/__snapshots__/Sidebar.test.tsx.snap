// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-AxjAm sc-fznyAO heCqGK"
  width="100%"
>
  <hr
    className="sc-AxirZ lcUwot"
    height="1px"
    width="100%"
  />
  <ul
    className="sc-fzplWN gsbudo"
    id="itens-sidebar"
  >
    <li
      className="sc-fzoLsD gSVrNH"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        className="sc-AxhCb eMREeU sc-fzpans hLMgdB"
        href="/dashboard"
        onClick={[Function]}
      >
        <div
          className="sc-AxjAm sc-AxgMl kwJSse"
          color="primary"
        >
          <svg
            fill="currentColor"
            height={18}
            size={18}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={18}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
        </div>
        <div
          className="sc-AxjAm fTjgAt"
          onClick={[Function]}
        >
          <div
            className="sc-AxjAm iTWSGy"
          >
            <span
              className="sc-AxiKw fYlGdG"
              color="darkerGrey"
              cursor="pointer"
            >
              dashboard
            </span>
          </div>
        </div>
      </a>
      <div
        className="sc-AxjAm sc-fzozJi fBHEQz hide"
      />
    </li>
    <li
      className="sc-fzoLsD gSVrNH"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        className="sc-AxhCb eMREeU sc-fzpans hLMgdB"
        href="/pedidos"
        onClick={[Function]}
      >
        <div
          className="sc-AxjAm sc-AxgMl kwJSse"
          color="primary"
        >
          <svg
            fill="currentColor"
            height={18}
            size={18}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={18}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </div>
        <div
          className="sc-AxjAm fTjgAt"
          onClick={[Function]}
        >
          <div
            className="sc-AxjAm iTWSGy"
          >
            <span
              className="sc-AxiKw fYlGdG"
              color="darkerGrey"
              cursor="pointer"
            >
              bookmark
            </span>
          </div>
        </div>
      </a>
      <div
        className="sc-AxjAm sc-fzozJi fBHEQz hide"
      />
    </li>
    <li
      className="sc-fzoLsD gSVrNH"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxjAm fvINga"
        onClick={[Function]}
        width="100%"
      >
        <div
          className="sc-AxjAm sc-AxgMl kwJSse"
          color="primary"
        >
          <svg
            fill="currentColor"
            height={18}
            size={18}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={18}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 20h20v-4H2v4zm2-3h2v2H4v-2zM2 4v4h20V4H2zm4 3H4V5h2v2zm-4 7h20v-4H2v4zm2-3h2v2H4v-2z"
            />
          </svg>
        </div>
        <div
          className="sc-AxjAm fTjgAt"
          onClick={[Function]}
        >
          <div
            className="sc-AxjAm iTWSGy"
          >
            <span
              className="sc-AxiKw fYlGdG"
              color="darkerGrey"
              cursor="pointer"
            >
              store
            </span>
          </div>
          <span
            className="sc-AxiKw sc-AxmLO IyyHK"
          >
            18
          </span>
          <div
            className="sc-AxjAm sc-AxgMl jhFwNa"
            color="primary"
          >
            <svg
              fill="currentColor"
              height={18}
              size={18}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="sc-AxjAm sc-fzozJi fBHEQz hide"
      >
        <ul>
          <li>
            <a
              className="sc-AxhCb eMREeU"
              href="/subItem"
              onClick={[Function]}
            >
              <span
                className="sc-AxiKw BfAh"
                color="darkGrey"
                cursor="pointer"
                fontSize="medium"
              >
                Sub Item
              </span>
            </a>
          </li>
          <li>
            <a
              className="sc-AxhCb eMREeU"
              href="/subItem"
              onClick={[Function]}
            >
              <span
                className="sc-AxiKw BfAh"
                color="darkGrey"
                cursor="pointer"
                fontSize="medium"
              >
                Sub Item
              </span>
            </a>
          </li>
          <li>
            <a
              className="sc-AxhCb eMREeU"
              href="/subItem"
              onClick={[Function]}
            >
              <span
                className="sc-AxiKw BfAh"
                color="darkGrey"
                cursor="pointer"
                fontSize="medium"
              >
                Sub Item
              </span>
              <span
                className="sc-AxiKw sc-AxmLO IyyHK"
              >
                18
              </span>
            </a>
          </li>
        </ul>
      </div>
    </li>
    <li
      className="sc-fzoLsD gSVrNH"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxjAm fvINga"
        onClick={[Function]}
        width="100%"
      >
        <div
          className="sc-AxjAm sc-AxgMl kwJSse"
          color="primary"
        >
          <svg
            fill="currentColor"
            height={18}
            size={18}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={18}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 20h20v-4H2v4zm2-3h2v2H4v-2zM2 4v4h20V4H2zm4 3H4V5h2v2zm-4 7h20v-4H2v4zm2-3h2v2H4v-2z"
            />
          </svg>
        </div>
        <div
          className="sc-AxjAm fTjgAt"
          onClick={[Function]}
        >
          <div
            className="sc-AxjAm iTWSGy"
          >
            <span
              className="sc-AxiKw fYlGdG"
              color="darkerGrey"
              cursor="pointer"
            >
              itens
            </span>
          </div>
          <div
            className="sc-AxjAm sc-AxgMl jhFwNa"
            color="primary"
          >
            <svg
              fill="currentColor"
              height={18}
              size={18}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={18}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="sc-AxjAm sc-fzozJi fBHEQz hide"
      >
        <ul>
          <li>
            <a
              className="sc-AxhCb eMREeU"
              href="/subItem"
              onClick={[Function]}
            >
              <span
                className="sc-AxiKw BfAh"
                color="darkGrey"
                cursor="pointer"
                fontSize="medium"
              >
                Sub Item
              </span>
            </a>
          </li>
          <li>
            <a
              className="sc-AxhCb eMREeU"
              href="/subItem"
              onClick={[Function]}
            >
              <span
                className="sc-AxiKw BfAh"
                color="darkGrey"
                cursor="pointer"
                fontSize="medium"
              >
                Sub Item
              </span>
            </a>
          </li>
          <li>
            <a
              className="sc-AxhCb eMREeU"
              href="/subItem"
              onClick={[Function]}
            >
              <span
                className="sc-AxiKw BfAh"
                color="darkGrey"
                cursor="pointer"
                fontSize="medium"
              >
                Sub Item
              </span>
            </a>
          </li>
          <li>
            <a
              className="sc-AxhCb eMREeU"
              href="/subItem"
              onClick={[Function]}
            >
              <span
                className="sc-AxiKw BfAh"
                color="darkGrey"
                cursor="pointer"
                fontSize="medium"
              >
                Sub Item
              </span>
            </a>
          </li>
          <li>
            <a
              className="sc-AxhCb eMREeU"
              href="/subItem"
              onClick={[Function]}
            >
              <span
                className="sc-AxiKw BfAh"
                color="darkGrey"
                cursor="pointer"
                fontSize="medium"
              >
                Sub Item
              </span>
            </a>
          </li>
          <li>
            <a
              className="sc-AxhCb eMREeU"
              href="/subItem"
              onClick={[Function]}
            >
              <span
                className="sc-AxiKw BfAh"
                color="darkGrey"
                cursor="pointer"
                fontSize="medium"
              >
                Sub Item
              </span>
            </a>
          </li>
        </ul>
      </div>
    </li>
    <li
      className="sc-fzoLsD gSVrNH"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        className="sc-AxhUy dzdgDV sc-fzpans hLMgdB"
        href="https://aiqfome.com/"
      >
        <div
          className="sc-AxjAm sc-AxgMl kwJSse"
          color="primary"
        >
          <svg
            fill="currentColor"
            height={18}
            size={18}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={18}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </div>
        <div
          className="sc-AxjAm fTjgAt"
          onClick={[Function]}
        >
          <div
            className="sc-AxjAm iTWSGy"
          >
            <span
              className="sc-AxiKw fYlGdG"
              color="darkerGrey"
              cursor="pointer"
            >
              aiqfome
            </span>
          </div>
        </div>
      </a>
      <div
        className="sc-AxjAm sc-fzozJi fBHEQz hide"
      />
    </li>
  </ul>
</div>
`;
